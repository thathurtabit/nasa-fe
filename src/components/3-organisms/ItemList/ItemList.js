import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import uuidv1 from 'uuid/v1';
import { CSSTransition } from 'react-transition-group';
import ItemThumb from '../../2-molecules/ItemThumb/ItemThumb';
import ItemListStyled from './ItemList.styled';
import { loadDelay } from '../../../utils/constants/constants';

const ItemList = ({ items }) => (
  <Fragment>
    <CSSTransition classNames="items-in" appear timeout={loadDelay}>
      <ItemListStyled id="item-list" className="items-in">
        {items.map((item, index) => (
          <ItemThumb num={index} key={uuidv1()} item={item} />
        ))}
      </ItemListStyled>
    </CSSTransition>
  </Fragment>
);

export default ItemList;

ItemList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
};
