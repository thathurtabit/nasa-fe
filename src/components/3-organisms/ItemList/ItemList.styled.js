import styled from 'styled-components';

const ItemListStyled = styled.ul`
  align-items: center;
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 30px;
  justify-items: center;
  list-style-type: none;
  margin: 20px auto 100px;
  max-width: ${props => props.theme.itemList.maxWidth};
  padding: 0 40px;
  transform: translateY(20px);
  transition: opacity ${props => props.theme.transition.duration}
      ${props => props.theme.transition.easeOut},
    transform ${props => props.theme.transition.duration}
      ${props => props.theme.transition.easeOut};

  @media screen and (min-width: ${props => props.theme.breakpoints.sm}) {
    grid-template-columns: repeat(2, 1fr);
  }

  @media screen and (min-width: ${props => props.theme.breakpoints.md}) {
    grid-template-columns: repeat(3, 1fr);
  }

  @media screen and (min-width: ${props => props.theme.breakpoints.lg}) {
    grid-template-columns: repeat(4, 1fr);
  }

  @media screen and (min-width: ${props => props.theme.breakpoints.xl}) {
    grid-template-columns: repeat(5, 1fr);
  }
`;

export default ItemListStyled;
