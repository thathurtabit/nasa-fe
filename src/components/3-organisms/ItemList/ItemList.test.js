import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';
import { shallowWithTheme } from '../../../utils/test_config/testHelpers';

import ItemList from './ItemList';

describe('ItemList', () => {
  it('should render correctly', () => {
    const element = shallowWithTheme(<ItemList items={['string']} />);
    expect(shallowToJson(element)).toMatchSnapshot();
  });
});
