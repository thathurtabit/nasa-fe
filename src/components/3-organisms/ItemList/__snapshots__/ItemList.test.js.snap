// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
